lib/tasks/migrate_working_pattern.rake

```diff
@@ -0,0 +1,16 @@
+namespace :data do
+  desc 'Migrate vacancy working pattern'
+  namespace :working_pattern do
+    task migrate: :environment do
+      Rails.logger.debug("Running vacancy working pattern migration task in #{Rails.env}")
+
+      Vacancy.where.not(working_pattern: nil)
+             .each do |vacancy|
+        vacancy.working_patterns = [vacancy.working_pattern]
+        vacancy.working_pattern = nil
```

Comment:
I'm not confident in the consequences of removing this. Setting to `nil` is what guards against reruns on already migrated (and possibly edited) data.

I've tried to filter out the cases where `working_patterns` is nil or empty, which would make this safer. but I can't work out the incantation that actually works in practice. Any ideas?

Reply:
> but I can't work out the incantation that actually works in practice

Not sure I'd know off the top of my head either, but I'm happy to jump on a call together.

Reply:
If we can only include nil or empty `working_patterns`, I'm happy to remove the `'nil`ing of `working_pattern`.

I have tried `Vacancy.where(working_patterns: nil)...` but that doesn't seem to find any records in the tests, and I don't know how to specify an empty array, since `where` parses arrays as options.

Reply:
I'd be happier keeping the `working_pattern` value around too (for safety's sake), I reckon you could include the cases where `working_patterns` in `nil` or `empty` with ` Vacancy.where(working_patterns: [])`. I've just tested it out and it seems to work for vacancies where working patterns are `nil` or `[]`, but you can probably drive this out with tests

Reply:
Argh. That appears to work now. I was having a lot of problems with that. Thanks.

Reply:
This is now resolved 🎉! Thanks both.

Reply:
Awesome! I'm happy in that case 👍